Mathlib/Data/ENat/Basic.lean

```diff
@@ -96,13 +96,20 @@ instance : WellFoundedRelation ℕ∞ where
   rel := (· < ·)
   wf := IsWellFounded.wf
 
-/-- Conversion of `ℕ∞` to `ℕ` sending `∞` to `0`. -/
-def toNat : MonoidWithZeroHom ℕ∞ ℕ where
+/-- Homomorphism from `ℕ∞` to `ℕ` sending `∞` to `0`. -/
+def toNatHom : MonoidWithZeroHom ℕ∞ ℕ where
   toFun := WithTop.untop' 0
   map_one' := rfl
   map_zero' := rfl
   map_mul' := WithTop.untop'_zero_mul
-#align enat.to_nat ENat.toNat
+#align enat.to_nat ENat.toNatHom
+
+/-- Conversion of `ℕ∞` to `ℕ` sending `∞` to `0`. -/
+def toNat : ℕ∞ → ℕ := toNatHom
+
+@[simp]
+theorem toNatHom_eq_toNat : toNatHom n = toNat n :=
+  rfl
```

Comment:
```suggestion
@[simp, norm_cast] lemma coe_toNatHom : toNatHom = toNat := rfl
lemma toNatHom_apply (n : ℕ) : toNatHom n = toNat n := rfl
```

Reply:
Btw, `by norm_cast` can prove `coe_toNatHom`, is it okay to give it `norm_cast` attribute if it is solved by it?